File: app/src/main/java/com/example/networkaplication/persistance/repository/database/AppDatabase.java
package com.example.networkaplication.persistance.repository.database;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;

import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;
import android.support.annotation.NonNull;

import com.example.networkaplication.persistance.model.Details;
import com.example.networkaplication.persistance.model.MovieQuery;
import com.example.networkaplication.persistance.repository.database.dao.DetailsDao;
import com.example.networkaplication.persistance.repository.database.dao.MovieQueryDao;

@Database(version = 4, exportSchema = false, entities = {MovieQuery.class, Details.class})
public abstract class AppDatabase extends RoomDatabase {

    static final String NAME = "app_database.db";

    public abstract MovieQueryDao movieQueryDao();

    public abstract DetailsDao detailsDao();

    public static synchronized AppDatabase build(@NonNull Context context) {
        return Room.databaseBuilder(context, AppDatabase.class, AppDatabase.NAME)
                .addCallback(new RoomDatabase.Callback() {
                    @Override
                    public void onCreate(@NonNull SupportSQLiteDatabase db) {

                    }

                    @Override
                    public void onOpen(@NonNull SupportSQLiteDatabase db) {
                    }
                })
                .fallbackToDestructiveMigration()
                .allowMainThreadQueries()
                .build();
    }
}

File: app/src/main/java/com/example/networkaplication/persistance/repository/database/AppDatabase.kt
package com.example.networkaplication.persistance.repository.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.sqlite.db.SupportSQLiteDatabase

import com.example.networkaplication.persistance.model.Details
import com.example.networkaplication.persistance.model.MovieQuery
import com.example.networkaplication.persistance.repository.database.dao.DetailsDao
import com.example.networkaplication.persistance.repository.database.dao.MovieQueryDao

@Database(version = 4, exportSchema = false, entities = [MovieQuery::class, Details::class])
abstract class AppDatabase : RoomDatabase() {

    abstract fun movieQueryDao(): MovieQueryDao

    abstract fun detailsDao(): DetailsDao

    companion object {

        private const val NAME = "app_database.db"

        @Synchronized
        fun build(context: Context): AppDatabase {
            return Room.databaseBuilder(context, AppDatabase::class.java, NAME)
                    .addCallback(object : RoomDatabase.Callback() {
                        override fun onCreate(db: SupportSQLiteDatabase) {

                        }

                        override fun onOpen(db: SupportSQLiteDatabase) {}
                    })
                    .fallbackToDestructiveMigration()
                    .allowMainThreadQueries()
                    .build()
        }
    }
}